phase1/gce/configure-vm-kubeadm.sh

```diff
@@ -27,13 +30,38 @@ else
   exit 1
 fi
 
+if [[ "${ENABLE_CLOUD_PROVIDER}" == true ]]; then
+  cat <<EOF > /etc/systemd/system/kubelet.service.d/20-cloud-provider.conf
+[Service]
+Environment="KUBELET_EXTRA_ARGS=--cloud-provider=${CLOUD_PROVIDER}"
+EOF
+
+  systemctl daemon-reload
+  systemctl restart kubelet
+fi
+
 case "${ROLE}" in
   "master")
-    OPTS=''
-    if [[ -n "$KUBERNETES_VERSION" ]]; then
-      OPTS="--kubernetes-version $KUBERNETES_VERSION"
+      cat <<EOF > kubeadm.conf
+kind: MasterConfiguration
```

Comment:
As a heads-up, this section is being reworked as part of upgrade functionality being added here:

https://github.com/kubernetes/kubernetes-anywhere/pull/450/files#diff-990f027176aa91f2d9cdff0b6278c1bdR65

That change is to persist the flags used during `kubeadm init` so that we can reuse them later when upgrading a cluster. I assume that using the config file instead is really our only option to support settings like `controllerManagerExtraArgs`, but that is going to become another merge conflict soon.